File: engine/model/src/main/scala/net/server/channel/packet/cash/use/UseItemMegaphonePacket.scala
package net.server.channel.packet.cash.use

class UseItemMegaphonePacket(private var _position: Short, private var _itemId: Int, private var _message: String, private var _whisper: Boolean, private var _selected: Boolean, private var _inventoryType: Byte, private var _slot: Short) extends AbstractUseCashItemPacket(_position, _itemId) {
  def message: String = _message

  def whisper: Boolean = _whisper

  def selected: Boolean = _selected

  def inventoryType: Byte = _inventoryType

  def slot: Short = _slot
}

File: engine/engine-model/src/main/java/net/server/channel/packet/cash/use/UseItemMegaphonePacket.java
package net.server.channel.packet.cash.use;

public class UseItemMegaphonePacket extends AbstractUseCashItemPacket {
   private final String message;

   private final Boolean whisper;

   private final Boolean selected;

   private final Byte inventoryType;

   private final Short slot;

   public UseItemMegaphonePacket(Short position, Integer itemId, String message, Boolean whisper, Boolean selected, Byte inventoryType, Short slot) {
      super(position, itemId);
      this.message = message;
      this.whisper = whisper;
      this.selected = selected;
      this.inventoryType = inventoryType;
      this.slot = slot;
   }

   public String message() {
      return message;
   }

   public Boolean whisper() {
      return whisper;
   }

   public Boolean selected() {
      return selected;
   }

   public Byte inventoryType() {
      return inventoryType;
   }

   public Short slot() {
      return slot;
   }
}
